src/dso_api/dynamic_api/permissions.py

```diff
@@ -58,19 +55,28 @@ def fetch_scopes_for_model(model: Type[Model]) -> TableScopes:
     return fetch_scopes_for_dataset_table(dataset, table)
 
 
-def get_unauthorized_fields(request, model) -> set:
-    """Check which field names should be excluded"""
+def get_unauthorized_fields(request, model) -> Set[str]:
+    """Returns a list of field names to be excluded from a response."""
+    # is_authorized_for is added by authorization_django middleware
+    # only if an authorization check is necessary.
+    if not hasattr(request, "is_authorized_for"):
+        return set()
+
     model_scopes = fetch_scopes_for_model(model)
     unauthorized_fields = set()
-    # is_authorized_for is added by authorization_django middleware
-    if hasattr(request, "is_authorized_for"):
-        for model_field in model._meta.get_fields():
-            scopes = model_scopes.fields.get(model_field.name)
-            scopes = scopes.union(model_scopes.table) if scopes else model_scopes.table
+
+    for model_field in model._meta.get_fields():
+        name = model_field.name
```

Comment:
Is this temporary variable really needed? It's used in 2 places.

Reply:
It makes the following code a bit less dense.

Reply:
Can it be named clearly? `field_name` is better, but then again. It's just few characters saved for 2 occasions.

Reply:
I've removed it :)